test/e2eexternal/dataplane/connectivity.go

```diff
@@ -0,0 +1,335 @@
+	"fmt"
+	"io"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var _ = Describe("[external][dataplane] Connectivity", func() {
+	f := framework.NewFramework("ext-dataplane")
+
+	It("Should be able to connect from external app to pod in cluster", func() {
+		testExternalConnectivity(f)
+	})
+})
+
+const (
+	extAppName = "ext-app"
+	extNetName = "pseudo-ext"
+
+	testImage         = "registry.access.redhat.com/ubi7/ubi:latest"
+	testContainerName = "ext-test-container"
+	testLabelkey      = "ext-test"
+)
+
+var (
+	podNameBase             = "ext-test-pod"
+	svcName                 = "ext-test-svc"
+	simpleHTTPServerCommand = []string{"python", "-m", "SimpleHTTPServer", "80"}
+	testLabel               = map[string]string{testLabelkey: podNameBase}
+)
+
+func testExternalConnectivity(f *framework.Framework) {
+	for idx, cs := range framework.KubeClients {
+		By(fmt.Sprintf("Creating a pod in cluster %q", framework.TestContext.ClusterIDs[idx]))
+		var err error
+
+		pod := makePodSpec(podNameBase, testLabel, testContainerName, testImage, simpleHTTPServerCommand)
+
+		pod, err = createPod(cs, f.Namespace, pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait for the pod to be ready
+		err = waitForPodReady(cs, f.Namespace, pod.Name)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Get the pod again to update after the pod become ready (To use pod.Status.PodIP, etc..)
+		pod, err = cs.CoreV1().Pods(f.Namespace).Get(pod.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Creating a service for the pod in cluster %q", framework.TestContext.ClusterIDs[idx]))
+
+		svc := makeServiceSpec(svcName, testLabel, 80, 80)
+
+		svc, err = createService(cs, f.Namespace, svc)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Get IPs to use later
+		podIP := pod.Status.PodIP
+		svcIP := svc.Spec.ClusterIP
+		dockerIP, err := getDockerIP(extAppName, extNetName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Sending http request from external app %q to service %q in cluster %q",
+			dockerIP, svcIP, framework.TestContext.ClusterIDs[idx]))
+
+		err = sendHTTPRequestFromDocker(extAppName, svcIP, 80)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verifying that pod get request")
+
+		podLog, err := getLogFromPod(cs, f.Namespace, pod.Name, testContainerName)
+
+		Expect(err).NotTo(HaveOccurred())
+		// TODO: For cluster1 (cluster that is connected to external network),
+		// source IP seems to become the gateway IP of the cluster CIDR, like 10.1.1.1
+		// not the podIP. This needs to be investigated further, before merge.
```

Comment:
You say this needs to be investigated before merge :-)

I'm ok with "this needs to be improved later" lol, but it would look like if we didn't review the coded hahaha.

Let's do it or update the comment :-)